pkg/controller/jenkins/configuration/base/resources/route.go

```diff
@@ -0,0 +1,57 @@
+package resources
+
+import (
+	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
+	routev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+//RouteKind the kind name for route
+const RouteKind = "Route"
+//RouteAPIIsDiscoverable true when the Route API is installed and discoverable
+var RouteAPIIsDiscoverable = false
+
+func init ()  {
+	RouteAPIIsDiscoverable,_ = IsRouteAPIDiscoverable()
+
+}
+// UpdateRoute returns new route matching the service
+func UpdateRoute(actual routev1.Route,jenkins *v1alpha2.Jenkins) routev1.Route {
+	actualTargetService := actual.Spec.To
+	serviceName := GetJenkinsHTTPServiceName(jenkins)
+	if( actualTargetService.Name != serviceName ) {
+		actual.Spec.To.Name = serviceName
+	}
+	port := jenkins.Spec.Service.Port
+	if( actual.Spec.Port.TargetPort.IntVal != port){
+		actual.Spec.Port.TargetPort = intstr.FromInt(int(port))
+	}
+	return actual
+}
+
+//IsRouteAPIDiscoverable tells if the Route API is installed and discoverable
+func IsRouteAPIDiscoverable() (bool, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		// Unable to get KBS Config
+		return false, err
+	}
+	k8s, err := kubernetes.NewForConfig(cfg)
```

Comment:
is already created here https://github.com/jenkinsci/kubernetes-operator/blob/master/cmd/manager/main.go#L118

Reply:
Changed the logic